chipflow_lib/steps/__init__.py

```diff
@@ -36,3 +40,26 @@ def build_cli_parser(self, parser):
     def run_cli(self, args):
         "Called when this step's is used from `chipflow` command"
         self.build()
+
+
+def _wire_up_ports(m: Module, top, platform):
+    for n, t in top.items():
+        setattr(m.submodules, n, t)
+
+    for component, iface in platform._pinlock.port_map.items():
+        for iface_name, member, in iface.items():
+            for name, port in member.items():
+                iface = getattr(top[component], iface_name)
+                wire = (iface if isinstance(iface.signature, IOSignature)
+                        else getattr(iface, name))
+                if port.invert:
+                    inv_mask = sum(inv << bit for bit, inv in enumerate(port.invert))
+                else:
+                    inv_mask = 0
+                port = platform._ports[port.port_name]
+                if hasattr(wire, 'i'):
+                    m.d.comb += wire.i.eq(port.i ^ inv_mask)
+                for d in ['o', 'oe']:
+                    if hasattr(wire, d):
+                        m.d.comb += getattr(port, d).eq(getattr(wire, d) ^ inv_mask)
```

Comment:
`oe` is never inverted regardless of `port.invert`, `invert` only refers to the input/output data